src/avram/associations/has_one.cr

```diff
@@ -11,7 +11,7 @@ module Avram::Associations::HasOne
     {% end %}
 
     {% unless foreign_key %}
-      {% foreign_key = "#{@type.name.underscore}_id".id %}
+      {% foreign_key = "#{@type.name.underscore.split("::").last.id}_id".id %}
```

Comment:
This was fixed under the [has_many](https://github.com/luckyframework/avram/blame/1a9018d9d37015c775404326ed1f33d9246554fd/src/avram/associations/has_many.cr#L9) a long time ago, so this now fixes it for the has_one